packages/autorest.python/autorest/codegen/serializers/sample_serializer.py

```diff
@@ -0,0 +1,144 @@
+        imports = FileImport()
+        namespace = (self.code_model.options["package_name"] or "").replace(
+            "-", "."
+        ) or self.code_model.namespace
+        imports.add_submodule_import(
+            namespace, self.code_model.client.name, ImportType.THIRDPARTY
+        )
+        credential_type = getattr(self.code_model.credential, "type", None)
+        if isinstance(credential_type, TokenCredentialType):
+            imports.add_submodule_import(
+                "azure.identity", "DefaultAzureCredential", ImportType.THIRDPARTY
+            )
+        elif isinstance(credential_type, AzureKeyCredentialType):
+            imports.add_import("os", ImportType.STDLIB)
+            imports.add_submodule_import(
+                "azure.core.credentials", "AzureKeyCredential", ImportType.THIRDPARTY
+            )
+        return FileImportSerializer(imports, True)
+
+    def _client_params(self) -> Dict[str, Any]:
+        # client params
+        special_param = dict()
+        credential_type = getattr(self.code_model.credential, "type", None)
+        if isinstance(credential_type, TokenCredentialType):
+            special_param.update({"credential": "DefaultAzureCredential()"})
+        elif isinstance(credential_type, AzureKeyCredentialType):
+            special_param.update(
+                {"credential": 'AzureKeyCredential(key=os.getenv("AZURE_KEY"))'}
+            )
+
+        params_positional = [
+            p
+            for p in self.code_model.client.parameters.positional
+            if not (p.optional or p.client_default_value)
+        ]
+        cls = lambda x: f'"{x}"'
+        client_params = {
+            p.client_name: special_param.get(
+                p.client_name,
+                cls(
+                    self.sample["parameters"].get(p.rest_api_name)
+                    or p.client_name.upper()
+                ),
+            )
+            for p in params_positional
+        }
+        if self.code_model.options["multiapi"]:
+            client_params["api_version"] = cls(self.operation_group.api_versions[0])
+
+        return client_params
+
+    # prepare operation parameters
+    def _operation_params(self) -> Dict[str, Any]:
+        params_positional = [
+            p
+            for p in self.operation.parameters.positional
+            if not p.client_default_value
+        ]
+        cls = lambda x: f'"{x}"' if isinstance(x, str) else str(x)
+        failure_info = '"fail to find required param named {%s} in example file {%s}"'
+        operation_params = {}
+        for param in params_positional:
+            name = param.rest_api_name
+            param_value = self.sample["parameters"].get(name)
+            if param_value or not param.optional:
+                if not param_value:
+                    # if can't find required param, need to log it
+                    _LOGGER.warning(failure_info, name, self.sample_origin_name)
```

Comment:
same comment as above: i think a lot of this code would be neater if we just loop over the parameters in the template and set values, there are a lot of cls etc code here, and it's pretty hard to read

Reply:
@iscai-msft parameters of examples are not always valid. They may be missing, may have different types. For example, when missing, we need to set a value by ourselves and log it; when different types(like int or string), we need add additional `"` to keep right. That's why we need this function.

Reply:
what do other languages do if the parameters are invalid? i'm not sure we should do a bunch of logic to fix it, that should be fixed in the examples

Reply:
.Net and Go has a bar that the generated sample at least should get compiled.
Shall we also add python linting check on the generated samples? We can abandon the samples if they are not qualified because of swagger sample problems, and this will require less sample-correction logics in generator.

CC to @weidongxu-microsoft

Reply:
@iscai-msft @changlong-liu Java has similar logic here with Python: log a warning if required parameter is missing in example files and go on.

Reply:
After discussion with @iscai-msft @changlong-liu  @weidongxu-microsoft, we will abandon the example with an log error and continue for other examples.
Update: done